pyuvdata/utils.py

```diff
@@ -3975,6 +4013,195 @@ def check_lsts_against_times(
         )
 
 
+def uvw_track_generator(
+    lon_coord=None,
+    lat_coord=None,
+    coord_frame="icrs",
+    coord_epoch=None,
+    coord_type="sidereal",
+    time_array=None,
+    telescope_loc=None,
+    antenna_frame="itrs",
+    antenna_positions=None,
+    antenna_numbers=None,
+    ant_1_array=None,
+    ant_2_array=None,
+    uvw_array=None,
+    force_postive_u=False,
+):
+    """
+    Calculate uvw coordinates (among other values) for a given position on the sky.
+
+    This function is meant to be a user-friendly wrapper around several pieces of code
+    for effectively simulating a track. Note that the naming of things may change
+    before this hits a PR.
+
+    Parameters
+    ----------
+    lon_coord : float or ndarray of float
+        Longitudinal (e.g., RA) coordinates, units of radians. Must match the same
+        shape as lat_coord.
+    lat_coord : float or ndarray of float
+        Latitudinal (e.g., Dec) coordinates, units of radians. Must match the same
+        shape as lon_coord.
+    coord_frame : string
+        The requested reference frame for the output coordinates, can be any frame
+        that is presently supported by astropy.
+    coord_epoch : float or str or Time object, optional
+        Epoch for ref_frame, nominally only used if converting to either the FK4 or
+        FK5 frames, in units of fractional years. If provided as a float and the
+        ref_frame is an FK4-variant, value will assumed to be given in Besselian
+        years (i.e., 1950 would be 'B1950'), otherwise the year is assumed to be
+        in Julian years.
+    time_array : float or ndarray of float or Time object
+        Times for which the apparent coordinates were calculated, in UTC JD. Must
+        match the shape of lon_coord and lat_coord.
```

Comment:
This confuses me. If I was phasing to a single RA/Dec I'd still expect to be able to pass a bunch of times to get my uvws for.

Reply:
I added the single-float case mostly because it was helpful for when I was testing/debugging (and there are instances where I can see getting a snapshot of UV-coverage would be a good thing). But in the interest of clarity,  I've taken it out of the docstring.